test/README.md

```diff
@@ -0,0 +1,107 @@
+# Writing Tests
+
+There are multiple flavors of node-addon-api test build that covers different
+build flags defined in `napi.h`:
+
+1. c++ exceptions enabled,
+2. c++ exceptions disabled,
+3. c++ exceptions disabled, and `NODE_ADDON_API_ENABLE_MAYBE` is defined.
+
+Functions in node-addon-api that call into JavaScript can have different
+declared return types to reflect build flavor settings. For example,
+`Napi::Object::Set` returns `bool` when `NODE_ADDON_API_ENABLE_MAYBE`
+is not defined, and `Napi::Maybe<bool>` when `NODE_ADDON_API_ENABLE_MAYBE`
+is defined. In source code, return type variants are defined as
+`Napi::MaybeOrValue<>` to prevent from duplicating most part of the code base.
+
+To properly test these build flavors, all values returned by a function defined
+with `Napi::MaybeOrValue<>` return types in node-addon-api test suite, should
+use one of the following test helpers to handle possible JavaScript exceptions.
+
+```cpp
+#include "napi.h"
+#include "test_helper.h"
+
+using namespace Napi;
+
+void fn(const CallbackInfo& info) {
+  Object obj = info[0].As<Object>();
+  Value value = MaybeUnwrap(obj->Get("foobar")); // <- `obj->Get` is calling
+  // into JavaScript and may throw JavaScript Exceptions. Here we just assert
+  // getting the parameter must not fail for convenience.
+
+  // ... do works with the value.
+}
+```
```

Comment:
I think this might be better after the three examples or maybe it could just be removed if we had examples in each of the sections for the helpers?